testsuite/kuadrant/policy/authorization/auth_policy.py

```diff
@@ -46,3 +45,24 @@ def add_rule(self, when: list["Rule"]):
         """Add rule for the skip of entire AuthPolicy"""
         self.model.spec.setdefault("when", [])
         self.model.spec["when"].extend([asdict(x) for x in when])
+
+    @property
+    def auth_section(self):
+        if self.spec_section is None:
+            self.spec_section = self.model.spec
+
+        spec_section = self.spec_section
+        self.spec_section = None
+        return spec_section.setdefault("rules", {})
+
+    @property
+    def defaults(self):
+        """Add new rule into the `defaults` AuthPolicy section"""
+        self.spec_section = self.model.spec.setdefault("defaults", {})
+        return self
+
+    @property
+    def overrides(self):
+        """Add new rule into the `overrides` AuthPolicy section"""
+        self.spec_section = self.model.spec.setdefault("overrides", {})
+        return self
```

Comment:
This is weird, while it indeed works, wont this cause weird artefacts like this?

```
authorization.identity.oidc() ->rules
authorization.overrides.identity.oidc() -> overrides
authorization.identity.oidc() -> overrides
```

Reply:
spec_section resets after new item is added. I have similar setup below 
```py
authorization.defaults.identity.add_oidc("inside-defaults", oidc_provider.well_known["issuer"])
authorization.identity.add_oidc("outside-defaults", oidc_provider.well_known["issuer"])
```

Reply:
That is awesome, thanks!